test/export/test_converter.py

```diff
@@ -869,6 +869,22 @@ def forward(self, x):
         #         )[0], 1
         #     )
 
+    def test_aten_tensor_dtype_int(self):
+        class M(torch.nn.Module):
+            def forward(self, x):
+                y = torch.tensor(1, dtype=torch.int32)
```

Comment:
Can we also test the following case

```
s = x.shape[0]
y = torch.tensor([s, s*2, 1])
```

Reply:
Updated with special casing depending on if the input is all constant ints or not. If it is all constant, we will replace with `torch.tensor` which will eventually create a constant tensor in the exported program. Otherwise, we will replace with `torch._ref.tensor` which will decompose into aten.scalar_tensor + aten._to_copy + aten.detach.